### ISSUE TITLE
NO_PROPOSAL_CHOSEN for inter-host communication

### ISSUE THREAD

**Issue by whs** [rancher/rancher#4987]
**Rancher Version:** 1.1.0-dev3

**Docker Version:** 1.10.3

**OS and where are the hosts located? (cloud, bare metal, etc):** CentOS 7 Bare metal, both in the same network

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)** Single master, external DB

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)** Cattle

I have two hosts located in the same network, 192.168.0.20 (also the rancher master) and 192.168.0.22. They both can ping each other successfully from the network container and has the correct IP shown in the web interface. This issue also happen in the 1.0.0 release.

#### Host  1

```
Jun  2 03:23:15 00[DMN] signal of type SIGHUP received. Reloading configuration
Jun  2 03:23:15 00[CFG] loaded 0 entries for attr plugin configuration
Jun  2 03:23:16 04[CFG] loaded IKE shared key for: '%any'
Jun  2 03:23:16 15[CFG] loaded IKE shared key for: '192.168.0.22'
Jun  2 03:23:16 09[CFG] replaced vici connection: conn-192.168.0.22
Jun  2 03:23:16 09[CFG] initiating 'child-192.168.0.22'
Jun  2 03:23:16 09[IKE] <conn-192.168.0.22|3> initiating IKE_SA conn-192.168.0.22[3] to 192.168.0.22
Jun  2 03:23:16 09[ENC] <conn-192.168.0.22|3> generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(HASH_ALG) ]
Jun  2 03:23:16 09[NET] <conn-192.168.0.22|3> sending packet: from 172.17.0.5[500] to 192.168.0.22[500] (1028 bytes)
Jun  2 03:23:16 05[CFG] loaded IKE shared key for: '10.42.104.56'
Jun  2 03:23:16 11[CFG] loaded IKE shared key for: '172.17.0.5'
Jun  2 03:23:16 15[NET] <conn-192.168.0.22|3> received packet: from 192.168.0.22[500] to 172.17.0.5[500] (36 bytes)
Jun  2 03:23:16 15[ENC] <conn-192.168.0.22|3> parsed IKE_SA_INIT response 0 [ N(NO_PROP) ]
Jun  2 03:23:16 15[IKE] <conn-192.168.0.22|3> received NO_PROPOSAL_CHOSEN notify error
Jun  2 03:24:17 10[NET] <4> received packet: from 172.17.0.1[47985] to 172.17.0.5[500] (1164 bytes)
Jun  2 03:24:17 10[ENC] <4> parsed IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(HASH_ALG) N(REDIR_SUP) ]
Jun  2 03:24:17 10[IKE] <4> no IKE config found for 172.17.0.5...172.17.0.1, sending NO_PROPOSAL_CHOSEN
Jun  2 03:24:17 10[ENC] <4> generating IKE_SA_INIT response 0 [ N(NO_PROP) ]
Jun  2 03:24:17 10[NET] <4> sending packet: from 172.17.0.5[500] to 172.17.0.1[47985] (36 bytes)
Jun  2 03:24:53 10[KNL] creating acquire job for policy 10.42.181.15/32[tcp] === 10.42.125.203/32[tcp/nameserver] with reqid {1234}
Jun  2 03:24:53 10[CFG] trap not found, unable to acquire reqid 1234
Jun  2 03:27:38 10[KNL] creating delete job for CHILD_SA ESP/0x00000000/192.168.0.22
Jun  2 03:27:38 13[JOB] CHILD_SA ESP/0x00000000/192.168.0.22 not found for delete
Jun  2 03:27:38 15[KNL] creating acquire job for policy 10.42.181.15/32[tcp/47867] === 10.42.125.203/32[tcp/nameserver] with reqid {1234}
```

```
root@a4609e2629f4:/# swanctl --list-conns
plugin 'sshkey' failed to load: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.22' not found (required by /usr/local/lib/ipsec/plugins/libstrongswan-sshkey.so)
conn-192.168.0.22: 
  local:  %any
  remote: 192.168.0.22
  local pre-shared key authentication:
  remote pre-shared key authentication:
  child-192.168.0.22: TUNNEL
    local:  0.0.0.0/0
    remote: 0.0.0.0/0

```

#### host 2

```
Jun  2 03:24:14 00[DMN] Starting IKE charon daemon (strongSwan 5.4.0, Linux 3.10.0-327.13.1.el7.x86_64, x86_64)
Jun  2 03:24:14 00[LIB] plugin 'sshkey' failed to load: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.22' not found (required by /usr/local/lib/ipsec/plugins/libstrongswan-sshkey.so)
Jun  2 03:24:15 00[CFG] loading ca certificates from '/usr/local/etc/ipsec.d/cacerts'
Jun  2 03:24:15 00[LIB] opening directory '/usr/local/etc/ipsec.d/cacerts' failed: No such file or directory
Jun  2 03:24:15 00[CFG]   reading directory failed
Jun  2 03:24:15 00[CFG] loading aa certificates from '/usr/local/etc/ipsec.d/aacerts'
Jun  2 03:24:15 00[LIB] opening directory '/usr/local/etc/ipsec.d/aacerts' failed: No such file or directory
Jun  2 03:24:15 00[CFG]   reading directory failed
Jun  2 03:24:15 00[CFG] loading ocsp signer certificates from '/usr/local/etc/ipsec.d/ocspcerts'
Jun  2 03:24:15 00[LIB] opening directory '/usr/local/etc/ipsec.d/ocspcerts' failed: No such file or directory
Jun  2 03:24:15 00[CFG]   reading directory failed
Jun  2 03:24:15 00[CFG] loading attribute certificates from '/usr/local/etc/ipsec.d/acerts'
Jun  2 03:24:15 00[LIB] opening directory '/usr/local/etc/ipsec.d/acerts' failed: No such file or directory
Jun  2 03:24:15 00[CFG]   reading directory failed
Jun  2 03:24:15 00[CFG] loading crls from '/usr/local/etc/ipsec.d/crls'
Jun  2 03:24:15 00[LIB] opening directory '/usr/local/etc/ipsec.d/crls' failed: No such file or directory
Jun  2 03:24:15 00[CFG]   reading directory failed
Jun  2 03:24:15 00[CFG] loading secrets from '/usr/local/etc/ipsec.secrets'
Jun  2 03:24:15 00[CFG] opening secrets file '/usr/local/etc/ipsec.secrets' failed: No such file or directory
Jun  2 03:24:15 00[LIB] loaded plugins: charon aesni aes des rc2 sha2 sha1 md5 random nonce x509 revocation constraints pubkey pkcs1 pkcs7 pkcs8 pkcs12 pgp dnskey pem fips-prf gmp xcbc cmac hmac gcm attr kernel-netlink resolve socket-default stroke vici updown xauth-generic
Jun  2 03:24:15 00[JOB] spawning 16 worker threads
Jun  2 03:24:15 03[CFG] loaded IKE shared key for: '%any'
Jun  2 03:24:15 08[CFG] loaded IKE shared key for: '172.17.0.2'
Jun  2 03:24:15 06[CFG] loaded IKE shared key for: '192.168.0.20'
Jun  2 03:24:15 10[CFG] added vici connection: conn-192.168.0.20
Jun  2 03:24:15 10[CFG] initiating 'child-192.168.0.20'
Jun  2 03:24:15 10[IKE] <conn-192.168.0.20|1> initiating IKE_SA conn-192.168.0.20[1] to 192.168.0.20
Jun  2 03:24:17 10[ENC] <conn-192.168.0.20|1> generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(HASH_ALG) N(REDIR_SUP) ]
Jun  2 03:24:17 10[NET] <conn-192.168.0.20|1> sending packet: from 172.17.0.2[500] to 192.168.0.20[500] (1164 bytes)
Jun  2 03:24:17 07[CFG] loaded IKE shared key for: '10.42.181.15'
Jun  2 03:24:17 16[NET] <conn-192.168.0.20|1> received packet: from 192.168.0.20[500] to 172.17.0.2[500] (36 bytes)
Jun  2 03:24:17 16[ENC] <conn-192.168.0.20|1> parsed IKE_SA_INIT response 0 [ N(NO_PROP) ]
Jun  2 03:24:17 16[IKE] <conn-192.168.0.20|1> received NO_PROPOSAL_CHOSEN notify error
Jun  2 03:24:58 11[KNL] creating acquire job for policy 10.42.125.203/32[icmp/8] === 10.42.74.190/32[icmp/8] with reqid {1234}
Jun  2 03:24:58 07[CFG] trap not found, unable to acquire reqid 1234
Jun  2 03:27:43 02[KNL] creating delete job for CHILD_SA ESP/0x00000000/192.168.0.20
Jun  2 03:27:43 08[JOB] CHILD_SA ESP/0x00000000/192.168.0.20 not found for delete
```

```
root@1812a1e8aa38:/# swanctl --list-conn
plugin 'sshkey' failed to load: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.22' not found (required by /usr/local/lib/ipsec/plugins/libstrongswan-sshkey.so)
conn-192.168.0.20: 
  local:  %any
  remote: 192.168.0.20
  local pre-shared key authentication:
  remote pre-shared key authentication:
  child-192.168.0.20: TUNNEL
    local:  0.0.0.0/0
    remote: 0.0.0.0/0
```

**Comment by leodotcloud** [comment]
@whs : Can you please check if you are hitting this problem in your setup? https://github.com/rancher/rancher/issues/4825#issuecomment-225080389

**Comment by whs** [comment]
I have CONFIG_INET_XFRM_MODE_TRANSPORT=m in my configuration file.
It's the standard CentOS kernel.

Anyway, in our setup it seems that this issue only happen with the
development cluster and so we decided to move off Rancher for now.

**Comment by svensp** [comment]
We had this issue in our setup after adding a new host with a more recent docker version
- 1.11.3 on the old hosts
- 1.12.1 on the added host
Restarting the agent-instance container as in the referenced issue did nothing.
It was fixed when we upgraded the old hosts to 1.12.1 as well.
